element-connector/src/main/java/org/eclipse/californium/elements/RelaxedCorrelationContextMatcher.java

```diff
@@ -36,29 +36,19 @@ public boolean isResponseRelatedToRequest(CorrelationContext requestContext, Cor
 		return internalMatch(requestContext, responseContext);
 	}
 
+	@Override
+	public boolean isToBeSent(CorrelationContext messageContext, CorrelationContext connectorContext) {
+		return internalMatch(messageContext, connectorContext);
+	}
+
 	private final boolean internalMatch(CorrelationContext requestedContext, CorrelationContext availableContext) {
 		if (null == requestedContext) {
 			return true;
 		} else if (null == availableContext) {
 			return false;
 		}
 		if (requestedContext.get(DtlsCorrelationContext.KEY_SESSION_ID) != null) {
-			boolean match = requestedContext.get(DtlsCorrelationContext.KEY_SESSION_ID).equals(
-					availableContext.get(DtlsCorrelationContext.KEY_SESSION_ID))
-					&& requestedContext.get(DtlsCorrelationContext.KEY_CIPHER).equals(
-							availableContext.get(DtlsCorrelationContext.KEY_CIPHER));
-
-			LOGGER.log(
-					match ? Level.FINEST : Level.WARNING,
-					"(D)TLS session {0}, {1}",
-					new Object[] { requestedContext.get(DtlsCorrelationContext.KEY_SESSION_ID),
-							availableContext.get(DtlsCorrelationContext.KEY_SESSION_ID) });
-			LOGGER.log(
-					match ? Level.FINEST : Level.WARNING,
-					"(D)TLS cipher {0}, {1}",
-					new Object[] { requestedContext.get(DtlsCorrelationContext.KEY_CIPHER),
-							availableContext.get(DtlsCorrelationContext.KEY_CIPHER) });
-			return match;
+			return CorrelationContextUtil.match(getName(), KEYS, requestedContext, availableContext);
```

Comment:
Good question. It's copied from the old UdpMatcher.

	private boolean isResponseRelatedToRequest(final Exchange exchange, final CorrelationContext responseContext) {
		if (exchange.getCorrelationContext() == null) {
			// no correlation information available for request, thus any
			// additional correlation information available in the response is ignored
			return true;
		} else if (exchange.getCorrelationContext().get(DtlsCorrelationContext.KEY_SESSION_ID) != null) {
			// original request has been sent via a DTLS protected transport
			// check if the response has been received in the same DTLS session
			if (useStrictResponseMatching) {
				return isResponseStrictlyRelatedToDtlsRequest(exchange.getCorrelationContext(), responseContext);
			} else {
				return isResponseRelatedToDtlsRequest(exchange.getCorrelationContext(), responseContext);
			}
		} else {
			// compare message context used for sending original request to context
			// the response has been received in
			return exchange.getCorrelationContext().equals(responseContext);
		}

There are some ideas an the correlation context (see https://github.com/eclipse/californium/issues/142), and, though I don't fully understand the usage of the correlation context, I didn't want to change too much.